src/OgAccess.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\og\OgAccess.
+ */
+
+namespace Drupal\og;
+
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\user\EntityOwnerInterface;
+
+class OgAccess {
+
+  /**
+   * Administer permission string.
+   *
+   * @var string
+   */
+  const ADMINISTER_GROUP_PERMISSION = 'administer group';
+
+  /**
+   * Determines whether a user has a given privilege.
+   *
+   * All permission checks in OG should go through this function. This
+   * way, we guarantee consistent behavior, and ensure that the superuser
+   * and group administrators can perform all actions.
+   *
+   * @param $group_type
+   *   The entity type of the group.
+   * @param $group_id
+   *   The entity ID of the group.
+   * @param string $operation
+   *   The entity operation being checked for.
+   * @param $account
+   *   (optional) The account to check. Defaults to the current user.
+   * @param $skip_alter
+   *   (optional) If TRUE then user access will not be sent to other modules
+   *   using drupal_alter(). This can be used by modules implementing
+   *   hook_og_user_access_alter() that still want to use og_user_access(), but
+   *   without causing a recursion. Defaults to FALSE.
+   * @param $ignore_admin
+   *   (optional) When TRUE the specific permission is checked, ignoring the
+   *   "administer group" permission if the user has it. When FALSE, a user
+   *   with "administer group" will be granted all permissions.
+   *   Defaults to FALSE.
+   *
+   * @return bool
+   *   TRUE or FALSE if the current user has the requested permission.
+   *   NULL, if the given group isn't a valid group.
+   */
+  public static function userAccess($group_type, $group_id, $operation, $account = NULL, $skip_alter = FALSE, $ignore_admin = FALSE) {
+    $perm = &drupal_static(__FUNCTION__, []);
```

Comment:
>  If we have that yet.

We have this `og_invalidate_cache()` that should be converted. Anyway, baby steps ;)